### DISCUSSION TITLE
Expenses plugin: Reporting

### DISCUSSION THREAD

**Discussion by idemus** [kimai/Paid-plugins#213] [General]
Hi Kevin,

I've installed the plugin yesterday and I supposed that the reporting gets also extended by the expenses I've booked on projects/activities, e.g. travel costs.

But unfortunatelly there is no change in the reporting sheets.

Do I something wrong or isn't it included at all?

Best regards
Klaus

**Comment by kevinpapst** [comment]
What do you mean? 
This here: https://www.kimai.org/documentation/plugin-expenses.html#budgets ?

**Reply by idemus** [reply]
I've already activated this checkbox but in "Reporting" column "costs" is still 0.00 (e.g. for "Reporting >monthly report") for the project I've assigned the expense to.
![image](https://github.com/user-attachments/assets/dd53e4ac-e993-4244-9ed5-563d6c8104de)

![image](https://github.com/user-attachments/assets/03344437-0a6d-4c89-88b4-11bf2b7bf508)

**Reply by kevinpapst** [reply]
These reports are about timesheets exclusively

**Reply by idemus** [reply]
Could you imagine to create an overarching report that takes all costs and earnings into account regardless whether they come from timesheets or from expenses and results in a sum where I can see at a glance how much profit/loss my project makes.

The workaround would be to create e.g. an activity "travel costs" which only has internal costs which then would be already reported in the currently timesheet reportings.

**Reply by kevinpapst** [reply]
Thats what the projects reports are meant for:

<img width="1127" alt="Bildschirmfoto 2024-12-20 um 11 52 39" src="https://github.com/user-attachments/assets/70ef339b-1038-4a19-8233-d194349570a2" />

**Reply by EduNetEurope** [reply]
> The workaround would be to create e.g. an activity "travel costs" which only has internal costs which then would be already reported in the currently timesheet reportings.

@idemus 
This is our current workaround. As a non-profit, we do not generate revenue in the traditional sense of the term, but we still need to monitor our costs.

The type of overarching report you are suggesting would be very helpful for us as well.

I use the export view to obtain the totals and rely on them for manual calculations.

Kimai is a brilliant tool. Thanks to its flexibility, we have managed to represent most of our needs with the help of workarounds., which is a remarkable improvement over our previous Excel sheets.

**Comment by idemus** [comment]
But "monthly report" doesn't give me the costs maintained in expenses. See example above. Column costs is 0.00.

**Reply by kevinpapst** [reply]
Its in the revenue column

**Reply by idemus** [reply]
But expenses are costs not revenues? Nevertheless also the column "revenue" is empty (0,00 €).
![image](https://github.com/user-attachments/assets/7f74d9a9-c74a-48a0-b18f-852b6c269a2b)

with following expense:
![image](https://github.com/user-attachments/assets/81b8f685-635c-4aae-877c-5b1c298b5e68)

**Reply by kevinpapst** [reply]
Is it billable?

**Reply by idemus** [reply]
No it's not.

**Reply by kevinpapst** [reply]
So its working very differently from what you expect I guess.

Help me figure out a way that is better and not breaking current workflows.

**Reply by idemus** [reply]
I have the need to get the earnings for each customer/project including all expenses, billable or not.
If billable, the expense would be revenue and cost, if not billable, the expense would be cost only.
So e.g. 400€ travel expense which is not billable (costs only) should be subtracted from the project/customer revenue and should not be included in the invoice whereas another travel expense for another customer/project over e.g. 80€ is billable and should be neutral in the earnings of that customer/project, meaning that it is included in the invoice (revenue) and it is included as cost.

Another example: wage expenses.
We have an own project 'working time' where all internal employees create their timesheet entries for working hours, holidays and other absences.
I want to book all wage expenses for each employee on that project to get a total sum of wage costs.

Long term short. Every expense should be taken into account as cost for every customer/project where it belongs to.

The 'master report':
Sum up all earnings over all customers/projects and get one 'company earning' per month/year.

**Reply by idemus** [reply]
There is also the situation that travel expenses are only paid partially by customer. Maybe there is a chance to have a field total expense and a field billable expense.
When the expense is billable and the field billable is empty the complete expense will be booked as revenue otherwise only the amount which is given to the field billable expense is used for revenue.
